### ISSUE TITLE
Allow using color encoded in an attribute

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/jbrowse-components#1734]
The itemRgb/reserved field in bigbed is an example

It would be nice to have this functionality baked in for some cases, since it is difficult to make a color callback to properly express this. 

Note that BAM also has this, and is not currently supported by our color by tag system (example YC:Z:255,0,0)

**Comment by cmdcolin** [comment]
this was brought up by a user at pag2024, how they had a hard time figuring out how to get the 'standard' BED itemRgb field to be used by a color callback

**Comment by cmdcolin** [comment]
ended up being in a field called field8 instead of itemRgb also, which i suppose if it was changed, would be a breaking change

**Comment by sfederman** [comment]
It would be great to support itemRgb natively in jbrowse, but here is a hack I used cobbled from other info I've gathered here in the meantime:


`  {
    type: 'FeatureTrack',
    trackId: 'trackID_1',
    name: 'trackID_1',
    assemblyNames: ['GRCh38'],
    adapter: {
      type: 'BedTabixAdapter',
      bedGzLocation: {
        uri: 'data/trackID_1.bed.gz'
      },
      index: {
        location: {
          uri: 'data/trackID_1.bed.gz.tbi'
        }
      }
    },
    displays: [
      {
        type: "LinearBasicDisplay",
        displayId: "SetRGBColor_filt01",
        renderer: {
          type: "SvgFeatureRenderer",
          color1: "jexl:get(feature,'field8')"
        }
      }
    ]
  },`

**Comment by cmdcolin** [comment]
yep thats the code to use, glad you got that example posted @sfederman  

it could be that the field8 (just auto-named fields by column number...) gets renamed to itemRgb, and then we can make code so that any feature that has an itemRgb uses that value.

**Comment by lidijaberke** [comment]
Hi,
Here is another vote in support of this feature being implemented.

**Comment by cmdcolin** [comment]
We now look for feature.get('itemRgb'), feature.get('reserved'), and feature.get('field8') and auto-color by this if it looks like an 'r,g,b' text, and if no other custom color configuration exists in the file otherwise

This is just released on main branch right now buit might close for now
thanks everyone for their patience. 
will update when released, it includes the large #5468 GPU powered refactor

**Comment by cmdcolin** [comment]
note that it will look for this field names from

- BED tabix columnNames config slot manually configured on the adapter configuration in config.json
- a header line in bedtabix file e.g. '#chr start end ... itemRgb'
- heuristic 'looks like a BED12' file assignment of default field names or bigBed autosql
- if you have a gff3 file instead, you can make an itemRgb or reserved col9 there too

**Comment by cmdcolin** [comment]
(and as usual, you can still do jexl overrides of the color)
